### PR TITLE
Add docker example for setting environment variables at runtime

### PR DESCRIPTION
**Pull request by evenchange4**
Add docker example for setting environment variables at runtime.

----
UPDATE:

Here I made two dockerfiles:
1. `dockerfile`: the simplest way to run Next.js app in docker (173MB) [\[demo\]](https://with-universal-configuration-runtime-docker-bxkuprrsog.now.sh/)
2. `dockerfile.multistage`: use multi-stage builds to build a smaller docker image (85MB) [\[demo\]](https://with-universal-configuration-runtime-docker-azvmlppsvt.now.sh/)

### CODE REVIEW

**Comment by evenchange4** [comment]
Thanks for the feedback. :) 
You are right. Both are great solutions to set environment variables at runtime.

The main purpose of this PR is about using Next.js with docker. 

----

If there is `'next/config'` importing in my React component file, I can not integrate it with React-Storybook. This is the reason why I prefer the universal way (I can mock window object).

File: examples/with-universal-configuration-runtime-docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM node:9-slim AS builder
```
Comment (timneutkens) [inline examples/with-universal-configuration-runtime-docker/Dockerfile]: Lets use `mhart/alpine-node`

File: examples/with-universal-configuration-runtime-docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /app
+COPY package.json .
+RUN npm install
```
Comment (timneutkens) [inline examples/with-universal-configuration-runtime-docker/Dockerfile]: Lets use yarn, it's included in `mhart/alpine-node` by default, so no need to install it. Just:

`RUN yarn install`

Comment (timneutkens) [reply examples/with-universal-configuration-runtime-docker/Dockerfile]: Lets not do multi-stage builds right now, but just `EXPOSE 3000` `CMD yarn start` in this step.

The reason for this is that `now` doesn't support multi-stage yet, it's coming soon though.

**Review by timneutkens — CHANGES_REQUESTED** [review]
Nice example, made some comments 🙏

File: examples/with-universal-configuration-runtime-docker/Dockerfile (hunk 2 above)
Comment (evenchange4) [reply examples/with-universal-configuration-runtime-docker/Dockerfile]: Actually, `now` do support multi-stage build in OSS.

Comment (evenchange4) [reply examples/with-universal-configuration-runtime-docker/Dockerfile]: I put this information in README.

File: examples/with-universal-configuration-runtime-docker/Dockerfile (hunk 1 above)
Comment (evenchange4) [reply examples/with-universal-configuration-runtime-docker/Dockerfile]: Sure, the `mhart/alpine-node` image is smaller than official one.

**Comment by evenchange4** [comment]
I updated the example. There are two examples:
1. `dockerfile`: the simplest way to run Next.js app in docker [\[demo\]](https://with-universal-configuration-runtime-docker-bxkuprrsog.now.sh/)
2. `dockerfile.multistage`: use multi-stage builds to build a smaller docker image [\[demo\]](https://with-universal-configuration-runtime-docker-azvmlppsvt.now.sh/)

**Comment by evenchange4** [comment]
@tusbar I made another PR which doing similar work in `next/config` way. 🙌

**Pull request closed without merging**